src/libraries/System.Runtime.Extensions/tests/System/IO/RedundantSegmentsTests_Windows.cs

```diff
@@ -0,0 +1,832 @@
+        public void Unqualified_Redundant_EdgeCases(string original, string expected) => TestAll(original, expected);
+
+        [Theory]
+        [MemberData(nameof(MemberData_Unqualified_Redundant_DrivelessRoot_SingleDot))]
+        [MemberData(nameof(MemberData_Unqualified_Redundant_DrivelessRoot_DoubleDot))]
+        public void Unqualified_Redundant_DrivelessRoot(string original, string expected) => TestAll(original, expected);
+
+        [Theory]
+        [MemberData(nameof(MemberData_Unqualified_Redundant_DrivelessRoot_SingleDot_EdgeCases))]
+        [MemberData(nameof(MemberData_Unqualified_Redundant_DrivelessRoot_DoubleDot_EdgeCases))]
+        public void Unqualified_Redundant_DrivelessRoot_EdgeCases(string original, string expected) => TestAll(original, expected);
+
+        [Theory]
+        [MemberData(nameof(MemberData_Unqualified_Redundant_DriveRootless_SingleDot))]
+        [MemberData(nameof(MemberData_Unqualified_Redundant_DriveRootless_DoubleDot))]
+        [MemberData(nameof(MemberData_Unqualified_Redundant_DriveRootless_Combined))]
+        public void Unqualified_Redundant_DriveRootless(string original, string expected) => TestAll(original, expected);
+
+        [Theory]
+        [MemberData(nameof(MemberData_Unqualified_Redundant_DriveRootless_SingleDot_EdgeCases))]
+        [MemberData(nameof(MemberData_Unqualified_Redundant_DriveRootless_DoubleDot_EdgeCases))]
+        public void Unqualified_Redundant_DriveRootless_EdgeCases(string original, string expected) => TestAll(original, expected);
+
+        [Theory]
+        [MemberData(nameof(MemberData_Unqualified_Redundant_Prefix_DriveRootless_SingleDot))]
+        [MemberData(nameof(MemberData_Unqualified_Redundant_Prefix_DriveRootless_DoubleDot))]
+        [MemberData(nameof(MemberData_Unqualified_Redundant_Prefix_DriveRootless_Combined))]
+        public void Unqualified_Redundant_Prefix_DriveRootless(string original, string expected) => TestAll(original, expected);
+
+        [Theory]
+        [MemberData(nameof(MemberData_Unqualified_Redundant_Prefix_DriveRootless_SingleDot_EdgeCases))]
+        [MemberData(nameof(MemberData_Unqualified_Redundant_Prefix_DriveRootless_DoubleDot_EdgeCases))]
+        public void Unqualified_Redundant_Prefix_DriveRootless_EdgeCases(string original, string expected) => TestAll(original, expected);
+
+        #endregion
+
+        #endregion
+
+        #region Test data
+
+        private const string ServerShare = @"\\Server\Share\";
+        private const string UNCServerShare = @"UNC\Server\Share\";
+        private const string Prefix_Windows_Drive_Rootless = "C:";
+        private const string Prefix_Windows_Driveless_Root = @"\";
+        private const string Prefix_Windows_Drive_Root = Prefix_Windows_Drive_Rootless + Prefix_Windows_Driveless_Root;
+        private static readonly string[] DeviceAndExtendedPrefixes = new string[]
+        {
+            @"\\.\",
+            @"\\?\",
```

Comment:
Paths in the form of  `\\?\` and `\??\` should never change as they are, by definition, fully qualified. If you pass a path with `?` to Windows there is no concept of "relative" segments- they won't be eaten/skipped. It is also theoretically possible to have some device that needs `\.\`  and `\..\` to be retained, `?` allows you to do that. 

I don't think we should remove segments from `\\?\C:\.` as `\\?\C:\` is _not_ treated the same by Windows. No matter what you do we should be super explicit in the docs about the special behavior of `\\?\`.

`\\.\` paths should, however, eat down to `Path.GetPathRoot()`:

`\\.\C:\`
`\\.\MyDrive\`
`\\.\UNC\Server\Share\`

Reply:
@JeremyKuhne I though about this case, but then I thought: if the user is explicitly asking to remove paths from the path prefixed with `\??\` or `\\?\`, we probably should help them. It's only when they call `GetFullPath` or other APIs that they need to be considered already fully qualified.

But now that you mention that `.` and `..` could be needed by devices, and need to be retained, I'll make sure to adjust the code and the unit tests to avoid removing segments from paths with these prefixes. If users need to remove segments from these paths, they can check if the path starts with `\\?\` or `\??\`, in which case, they should call the `RemoveRedundantSegments` APIs that take a span, and pass their string sliced (removing the prefix).

Thanks for confirming this.

Reply:
By the way, `Path.GetPathRoot()` does some amazing magic that helped me ensure a lot of cases got pre-covered.

Reply:
The other main issue I had with manipulating them is that they are no longer equivalent. Everything else you'd get the same results using the path before and after removing relative segments. I think that is one of the contracts here- you get the same results before and after. That said, there is weirdness around traversing links on Unix that I've never really rationalized. 🤷‍♂️

Reply:
@JeremyKuhne I remember another reason why I decided to make the `RemoveRedundantSegments` methods to work with paths prefixed with `\\?\`:
the pre-existing System.Runtime.Extensions unit tests for `GetFullPath` are expecting the paths prefixed with `\\?\` to get normalized.

Removing the ability from `GetFullPath` from removing redundant segments and duplicate separators from paths prefixed with `\\?\` or `\??\` would be considered a breaking change, wouldn't it?

https://github.com/dotnet/runtime/blob/da10ba8ca7cd103e4c6137b91522a834bcead726/src/libraries/System.Runtime.Extensions/tests/System/IO/PathTests_Windows.cs#L532-L538

Reply:
I'm inclined to consider this a bug fix instead: Paths that begin with `\\?\` should not get modified by either `GetFullPath(string, string)` or `RemoveRedundantSegments`, in which case I would update these unit tests so when calling `GetFullPath(string, string)` with a path prefixed with `\\?\` would expect the same string.

Reply:
Emphasis on "duplicate separators". It seems all these pre-existing unit tests are failing because they only expect duplicate separators to be collapsed, but maybe even that should not be addresed for paths prefixed with `\\?\` and `\??\`.

Reply:
I think I understand what's happening. Those `GetFullPath` unit tests expect the `\\?\` paths to get combined with the base path. I will create a separate unit test method to verify extended paths only.

Reply:
I updated the old existing unit tests. I split the cases that get `\\?\` prefixed into separate unit tests with different outputs (unmodified, except when the paths get combined).